features/FEATURE_LWIP/lwip-interface/ppp_lwip.cpp

```diff
@@ -18,6 +18,14 @@
 #include "platform/FileHandle.h"
 #include "platform/mbed_poll.h"
 #include "events/EventQueue.h"
+#if defined(FEATURE_COMMON_PAL)
+#include "mbed_trace.h"
+#define TRACE_GROUP "LPPP"
+#else
+#define tr_debug(...) (void(0)) //dummies if feature common pal is not added
```

Comment:
This commit was deliberately sidestepping the question of "why is mbed trace hidden inside a feature" - we were just aware that it was and we didn't want to force that feature to be enabled for PPP.  We were just using the trace for local testing (substituting for the built-in lwip printf trace that consumed too much stack).

And is a library even supposed to enable a feature - would we be requiring the app to enable it? Certainly we do require the app to turn on COMMON_PAL themselves to use Nanostack; Nanostack doesn't turn it on itself.

So as it stands, if the app does enable the COMMON_PAL feature, then we get the mbed trace support, and if they don't we compile without trace.